src/app/courses/courses.component.html

```diff
@@ -16,12 +16,12 @@ <h2 class="header" i18n> Courses </h2>
     <ng-container matColumnDef="action">
       <mat-header-cell *matHeaderCellDef i18n> Action </mat-header-cell>
       <mat-cell *matCellDef="let element">
-          <a mat-raised-button color="primary" routerLink="add" routerLinkActive="active" i18n>Open</a>
-          <a mat-raised-button color="primary" routerLink="add" routerLinkActive="active" i18n>Resign</a>
-          <a mat-raised-button color="primary" routerLink="add" routerLinkActive="active" i18n>Manage</a>
-          <a mat-raised-button color="primary" routerLink="add" routerLinkActive="active" i18n>View Course</a>
-          <a mat-raised-button color="primary" routerLink="add" routerLinkActive="active" i18n>Credits</a>
-          <a mat-raised-button color="primary" routerLink="add" routerLinkActive="active" i18n>Progress</a>
+          <a mat-raised-button color="primary" routerLink="open/{{element.courseTitle}}" routerLinkActive="active" i18n>Open</a>
```

Comment:
{{element.coureTitle}} instead of this, it should take course id. Since courseTitle might be too long and may contains special characters which is not good for url address. @paulbert  and @lmmrssa What do you think about this ?

Reply:
Yes, please use the `_id` property instead of `courseTitle`